tardis/plasma/properties/ion_population.py

```diff
@@ -223,31 +224,39 @@ def _calculate_block_ids(phi):
         return calculate_block_ids_from_dataframe(phi)
 
     def calculate(self, phi, partition_function, number_density):
-        n_e_convergence_threshold = 0.05
-        n_electron = number_density.sum(axis=0)
-        n_electron_iterations = 0
-
-        while True:
+        if self._electron_densities is None:
```

Comment:
@wkerzendorf  Okay, If I make a `ElectronDensities` module like this under `plasma_input.py`
```
class ElectronDensities(ArrayInput):
    outputs = ('electron_densities',)
```


- First of all, If we are not supplying `electron_densities` from CMFGEN file (normal cases), then how will I set this value ?
The Logic I saw in `IonNumberDensity` , calculates both `electron_densities` and `ion_number_density`. And if I make a separate module
for it, then I will have to remove `electron_densities` from the output list of `IonNumberDensity`, because then it will giver error while
resolving variables(`_resolve_update_list` method)

- And then I think , it will resulting in changing some tests/fixtures which are dependent on electron_density, if above behaviour is changed.